index.js

```diff
@@ -228,6 +275,11 @@ var withApp = function (app, apiOptions) {
                             return
                         }
 
+                        if (!isCached && cacheConfig) {
+                            let dataSource = { ...retValue, ...req }
+                            cacheConfig.keys = getCacheKeys(dataSource, cacheConfig, req.context)
```

Comment:
let keep cachConfig as `immutable`. can you not extract the keys into a new variable?